### PR TITLE
Added Serverless documentation

### PR DESCRIPTION
**Pull request by DedsecKnight**
Changes made in this PR:
- Implementation of SendGrid service and Vanity service. 
- Added documentation on Docusaurus on how to setup a serverless service.
- Added documentation related to GraphQL infrastructure.

### CODE REVIEW

File: apps/documentation/docs/getting-started/serverless.md [hunk 1]
```diff
@@ -2,24 +2,286 @@
...
+### Directory Setup
 
-- Run `npm install`
-- Run `npm run dev`
+- Create a new folder with the name of the service inside the `services` directory
+  - e.g: If your service name is `email`, then create the new directory `services/email`.
+- Run the following command and follow the prompt:
+
+```
+npm init -w ./services/email
```
Comment (stevenmqnguyen) [inline apps/documentation/docs/getting-started/serverless.md]:
```suggestion
npm init --scope @acmutd -w ./services/email
```

File: apps/documentation/docs/getting-started/serverless.md [hunk 2]
```diff
@@ -2,24 +2,286 @@
...
+```
+
+:::important
+
+- When run the above command, or any command afterward, make sure to replace `email` with the name of your service.
+
+- When being prompted package name, make sure that it should be in the form of `@acmutd/<service_name>`
```
Comment (stevenmqnguyen) [inline apps/documentation/docs/getting-started/serverless.md]: Can make @acmutd scope by adding the scope to npm init

File: apps/documentation/docs/getting-started/serverless.md [hunk 3]
```diff
@@ -2,24 +2,286 @@
...
-If you are using Windows, running `npm run dev` may cause some errors. By default this project is setup to inspect server side logs and print them to the terminal window running the application. If there are any issues remove the `NODE_OPTIONS='--inspect'` from the scripts section of the `package.json` file for `npm run dev`.
 :::
 
-After running these steps you should be able to open up `http://localhost:3000` and view the leadership site!
+### Install Serverless Function Dependencies
+
+- To install the dependencies that will be used by Serverless Framework, run the following command:
+
+```
+npm i -w ./services/email -D serverless-iam-roles-per-function serverless-create-global-dynamodb-table serverless-offline serverless-prune-plugin
```
Comment (stevenmqnguyen) [inline apps/documentation/docs/getting-started/serverless.md]:
```suggestion
npm i -w @acmutd/email -D serverless-iam-roles-per-function serverless-create-global-dynamodb-table serverless-offline serverless-prune-plugin
```

Comment (stevenmqnguyen) [reply apps/documentation/docs/getting-started/serverless.md]: same with the other instances

File: apps/documentation/docs/getting-started/serverless.md [hunk 4]
```diff
@@ -2,24 +2,286 @@
...
+```
+
+- To install AWS Lambda related dependencies, run the following command:
+
+```
+npm i -w ./services/email aws-sdk aws-lambda
+```
+
+- To install Lambda Powertools, run the following command:
```
Comment (stevenmqnguyen) [inline apps/documentation/docs/getting-started/serverless.md]: Move to root package.json

File: apps/documentation/docs/getting-started/serverless.md [hunk 5]
```diff
@@ -2,24 +2,286 @@
...
+
+```
+npm i -w ./services/email @dazn/lambda-powertools-cloudwatchevents-client @dazn/lambda-powertools-correlation-ids @dazn/lambda-powertools-logger @dazn/lambda-powertools-pattern-basic @dazn/lambda-powertools-lambda-client @dazn/lambda-powertools-sns-client @dazn/lambda-powertools-sqs-client @dazn/lambda-powertools-dynamodb-client @dazn/lambda-powertools-kinesis-client
+```
+
+### Setup Linting
+
+- To install linting dependencies, run the following command:
```
Comment (stevenmqnguyen) [inline apps/documentation/docs/getting-started/serverless.md]: Only need to install in root package.json I believe. Might need to test with husky & vscode extension

File: apps/documentation/docs/getting-started/serverless.md [hunk 6]
```diff
@@ -2,24 +2,286 @@
...
+      - httpApi:
+          # This is endpoint for function
+          path: /email
+          # This is HTTP method
+          method: post
+```
+
+### Run the function locally
+
+- To run an emulator of the function run the following command:
```
Comment (stevenmqnguyen) [inline apps/documentation/docs/getting-started/serverless.md]: clarify how to run using turbo as well `npx turbo run sls:offline --scope=@acmutd/<project>` from root dir

File: apps/documentation/docs/getting-started/serverless.md [hunk 7]
```diff
@@ -2,24 +2,286 @@
...
+:::important
+If your serverless function is dependent on environment variables taken from Doppler, you will need to run the following command:
+
+```
+doppler run -- sls offline
+```
+
+:::
+
+- The function should now be accesible at `http://localhost:3000/email`
```
Comment (stevenmqnguyen) [inline apps/documentation/docs/getting-started/serverless.md]: Is there a way to configure the default port?

File: services/sendgrid/src/handler.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
+/* eslint @typescript-eslint/no-unsafe-assignment: 0 */
```
Comment (stevenmqnguyen) [inline services/sendgrid/src/handler.ts]: Need to figure out why we need to disable these

**Review by stevenmqnguyen — DISMISSED** [review]
Excellent job with documentation, very comprehensive and concise.

File: apps/documentation/docs/getting-started/serverless.md (hunk 1 above)
Comment (stevenmqnguyen) [reply apps/documentation/docs/getting-started/serverless.md]: also run the command `Monorepo: sync workspace features` in the VSCode command palette to update the file sidebar config with new package

File: apps/documentation/docs/getting-started/serverless.md [hunk 9]
```diff
@@ -263,22 +273,19 @@ functions:
...
+- To run an emulator of the function run the following command from service directory:
 
 ```
-sls offline
+npm run sls:offline
 ```
 
-:::important
-If your serverless function is dependent on environment variables taken from Doppler, you will need to run the following command:
+- Another way to run the service would be the following command:
```
Comment (stevenmqnguyen) [inline apps/documentation/docs/getting-started/serverless.md]: specify that npx turbo can only be run in root dir

**Review by stevenmqnguyen — CHANGES_REQUESTED** [review]
Looks solid, did you test pushing?

**Comment by DedsecKnight** [comment]
> Looks solid, did you test pushing?

If you mean test pushing to AWS, I can try but I am not sure if it is necessary because I believe most of the changes I made to resolve ESLint error are on the `tsconfig.json` and `.eslintrc.json` file.

**Review by stevenmqnguyen — CHANGES_REQUESTED** [review]
Need to add .lintstagedrc.json. You can copy from the graphql lambda. Lint staged needs this in each project to ensure it will use the closest eslint and prettier configs i.e. the one in the project, and not the root.

**Comment by stevenmqnguyen** [comment]
Need to add `.eslintignore`, `.prettierrc`, `.lintstagedrc.json` to docs.

File: services/sendgrid/src/handler.ts [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+export const main: SQSHandler = async (event, context, callback) => {
+  const { Records } = event;
+  const stepFunction = new aws.StepFunctions();
+  try {
+    await Promise.all(
+      Records.map(async (record) => {
+        const messageBody = JSON.parse(record.body);
+        const { taskToken } = messageBody;
+        const emailData = messageBody.payload as SendEmailConfig;
```
Comment (NickBurnett) [inline services/sendgrid/src/handler.ts]: it's probably this

Comment (DedsecKnight) [reply services/sendgrid/src/handler.ts]: not sure what u mean by that

Comment (NickBurnett) [reply services/sendgrid/src/handler.ts]: For the lint-ignore.

Comment (NickBurnett) [reply services/sendgrid/src/handler.ts]: Top line of the file.

Comment (DedsecKnight) [reply services/sendgrid/src/handler.ts]: oh I see what u mean. I already fixed that one. Not sure why it was still there though.

**Review by stevenmqnguyen — APPROVED** [review]

**Pull request merged**